src/rawchannel.cc

```diff
@@ -312,41 +330,103 @@ class RawChannel : public Nan::ObjectWrap
     v8::Local<v8::Context> context = info.GetIsolate()->GetCurrentContext();
     v8::Local<v8::Object> obj = Nan::To<Object>(info[0]).ToLocalChecked();
 
+    // CAN HS FRAME
+    // ---------------
     // TODO: Check for correct structure of message object
-    frame.can_id = obj->Get(context, id_symbol).ToLocalChecked()->ToUint32(context).ToLocalChecked()->Value();
 
-    if (obj->Get(context, ext_symbol).ToLocalChecked()->IsTrue())
-      frame.can_id |= CAN_EFF_FLAG;
+      frame.can_id = obj->Get(context, id_symbol).ToLocalChecked()->ToUint32(context).ToLocalChecked()->Value();
+      if (obj->Get(context, ext_symbol).ToLocalChecked()->IsTrue())
+        frame.can_id |= CAN_EFF_FLAG;
 
-    if (obj->Get(context, rtr_symbol).ToLocalChecked()->IsTrue())
-      frame.can_id |= CAN_RTR_FLAG;
+      if (obj->Get(context, rtr_symbol).ToLocalChecked()->IsTrue())
+        frame.can_id |= CAN_RTR_FLAG;
 
-    v8::Local<v8::Value> dataArg = obj->Get(context, data_symbol).ToLocalChecked();
+      v8::Local<v8::Value> dataArg = obj->Get(context, data_symbol).ToLocalChecked();
 
-    CHECK_CONDITION(node::Buffer::HasInstance(dataArg), "Data field must be a Buffer");
+      CHECK_CONDITION(node::Buffer::HasInstance(dataArg), "Data field must be a Buffer");
 
-    // Get frame data
-    frame.can_dlc = node::Buffer::Length(Nan::To<Object>(dataArg).ToLocalChecked());
-    memcpy(frame.data, node::Buffer::Data(Nan::To<Object>(dataArg).ToLocalChecked()), frame.can_dlc);
+      // Get frame data
+      frame.can_dlc = node::Buffer::Length(Nan::To<Object>(dataArg).ToLocalChecked());    
+      memcpy(frame.data, node::Buffer::Data(Nan::To<Object>(dataArg).ToLocalChecked()), frame.can_dlc);
 
-    { // set time stamp when sending data
-      struct timeval now;
-      if ( 0==gettimeofday(&now, 0)) {
-        Nan::Set(obj, tssec_symbol, Nan::New((int32_t)now.tv_sec));
-        Nan::Set(obj, tsusec_symbol, Nan::New((int32_t)now.tv_usec));
+      { // set time stamp when sending data
+        struct timeval now;
+        if ( 0==gettimeofday(&now, 0)) {
```

Comment:
no yoda style please, please check gettime() == 0

Reply:
and why are you setting the time when sending though?

Reply:
I didn't change this portion, it's the original send() function.
Check your source code, it was already in.